src/components/modals/modal/modal.tsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react'
+import { createPortal } from 'react-dom'
+
+import styles from './modal.module.css'
+
+interface ModalProps {
+  isDisplayed: boolean
+  children: React.ReactNode
+  handleClose(): void
+}
+
+export const Modal: React.FC<ModalProps> = ({
+  isDisplayed,
+  children,
+  handleClose,
+}) => {
+  const body = document.querySelector('body')
+  const portal = document.createElement('div')
+  const containerRef = React.useRef(null)
+
+  const handleWindowClick = (event: any) => {
```

Comment:
you can type here as `event: MouseEvent`

https://stackoverflow.com/questions/57515417/not-assignable-type-for-event-listener